programs/drift/src/math/amm_spread.rs

```diff
@@ -95,6 +95,59 @@ pub fn cap_to_max_spread(
     Ok((long_spread, short_spread))
 }
 
+pub fn calculate_long_short_vol_spread(
+    last_oracle_conf_pct: u64,
+    reserve_price: u64,
+    mark_std: u64,
+    oracle_std: u64,
+    long_inten: u64,
+    short_inten: u64,
+    volume_24h: u64,
+) -> DriftResult<(u64, u64)> {
+    // 1.6 * std
+    let market_avg_std_pct = oracle_std
+        .safe_add(mark_std)?
+        .safe_mul(PERCENTAGE_PRECISION_U64)?
+        .safe_div(2 * reserve_price)?;
+
+    let vol_spread: u64 = last_oracle_conf_pct.max(market_avg_std_pct / 2);
+    crate::dlog!(vol_spread, market_avg_std_pct);
+
+    let long_vol_spread_factor: u64 = long_inten
+        .safe_mul(PERCENTAGE_PRECISION_U64)?
+        .safe_div(max(volume_24h, 1))?
+        .clamp(
+            PERCENTAGE_PRECISION_U64 / 100,
+            16 * PERCENTAGE_PRECISION_U64 / 10,
+        );
+    let short_vol_spread_factor: u64 = short_inten
+        .safe_mul(PERCENTAGE_PRECISION_U64)?
+        .safe_div(max(volume_24h, 1))?
+        .clamp(
+            PERCENTAGE_PRECISION_U64 / 100,
+            16 * PERCENTAGE_PRECISION_U64 / 10,
+        );
```

Comment:
so before we just had volatility = larger spread with last_oracle_conf_pct - and now were incorporating long_intensity so as more people go long we have a larger long spread to reduce long/short imbalance? (and with all the max's we make the spread greater than or equal to spread v1)